nexus-repository-helm/src/main/java/org/sonatype/repository/helm/HelmFacet.java

```diff
@@ -0,0 +1,75 @@
+package org.sonatype.repository.helm;
+
+import org.sonatype.nexus.repository.Facet;
+import org.sonatype.nexus.repository.FacetSupport;
+import org.sonatype.nexus.repository.storage.Asset;
+import org.sonatype.nexus.repository.storage.Bucket;
+import org.sonatype.nexus.repository.storage.Component;
+import org.sonatype.nexus.repository.storage.StorageTx;
+import org.sonatype.repository.helm.internal.HelmAssetAttributePopulator;
+import org.sonatype.repository.helm.internal.util.HelmDataAccess;
+
+import static org.sonatype.nexus.repository.storage.AssetEntityAdapter.P_ASSET_KIND;
+import static org.sonatype.repository.helm.internal.AssetKind.HELM_PACKAGE;
+
+public abstract class HelmFacet
+    extends FacetSupport
+    implements Facet
+{
+  public abstract HelmDataAccess getHelmDataAccess();
+
+  public abstract HelmAssetAttributePopulator getHelmAssetAttributePopulator();
+
+  protected Asset findOrCreateAssetWithComponent(
+      final String assetPath,
+      final StorageTx tx,
+      final Bucket bucket,
+      final HelmAttributes chart)
+  {
+    Asset asset = getHelmDataAccess().findAsset(tx, bucket, assetPath);
+    if (asset == null) {
+      Component component = findOrCreateComponent(tx, bucket, chart);
+      asset = tx.createAsset(bucket, component);
+      asset.name(assetPath);
+      asset.formatAttributes().set(P_ASSET_KIND, HELM_PACKAGE.name());
+    }
+
+    getHelmAssetAttributePopulator().populate(asset.formatAttributes(), chart);
+
+    return asset;
+  }
+
+  protected Asset findOrCreateAssetWithFormat(
```

Comment:
`findOrCreateAssetWithFormat()` -> `findOrCreateAssetWithAttributes()`.
Naming sounds a bit confusing cause `format` is always `helm` for all assets/components in this repo